core/src/test/java/org/elasticsearch/index/query/AbstractQueryTestCase.java

```diff
@@ -562,7 +577,7 @@ protected QB copyQuery(QB query) throws IOException {
      * @return a new {@link QueryShardContext} based on the base test index and queryParserService
      */
     protected static QueryShardContext createShardContext() {
-        QueryShardContext queryCreationContext = new QueryShardContext(queryParserService);
+        QueryShardContext queryCreationContext = queryShardContext.clone();
```

Comment:
nit: feels slightly cleaner if we don't use the shared queryShardContext instance member ,  maybe have the deps as instance member and recreate a new QueryShardContext here passing in the deps instead (like we do with the query parse context?